### PR TITLE
Document httpapi and cliconf plugins

### LINKED ISSUE #37292: Network docs: ANSIBLE_EOS_USE_SESSIONS
##### ISSUE TYPE
 - Documentation Report

##### COMPONENT NAME
eos_

##### ANSIBLE VERSION
```
2.4
```

##### STEPS TO REPRODUCE
`ANSIBLE_EOS_USE_SESSIONS` is used to enable/disable config sessions
if it is enabled, a config sessions is created and pushed configuration is applied in candidate db
after issuing `commit` command the candidate config  applied in running db

```yaml
  environment:
    ANSIBLE_EOS_USE_SESSIONS: 0
```

* This existed in Ansible 2.4 (and possibly earlier)
* This only exists on EOS, there isn't an equivalent for other platforms
* it is applicable for all the eos declarative modules that use `load_config()` api
* Noticed this when looking at https://github.com/ansible/ansible/pull/37287/files

**Remaining questions**
1) How would you know if/when you should use this

@acozine This is not a priority for 2.5 the option existed in Ansible 2.4 and earlier
If you need extra information @ganeshrn and @gdpak can assist

### PR DESCRIPTION
**Pull request by Qalthos**
##### SUMMARY

Some of these plugins have options, and should be documented in the standard way.

Fixes #37292

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

httpapi
cliconf

### CODE REVIEW

File: docs/docsite/rst/plugins/cliconf.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
+.. _cliconf_plugins:
+
+Cliconf Plugins
+===============
+
+.. contents::
+   :local:
+   :depth: 2
+
+Cliconf plugins are abstactions over the cli interface to network devices. They provide a standard interface
```
Comment (samccann) [inline docs/docsite/rst/plugins/cliconf.rst]:
```suggestion
Cliconf plugins are abstactions over the CLI interface to network devices. They provide a standard interface
```

File: docs/docsite/rst/plugins/cliconf.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+for Ansible to execute tasks on them.
```
Comment (samccann) [inline docs/docsite/rst/plugins/cliconf.rst]:
```suggestion
for Ansible to execute tasks on those network devices.
```

File: docs/docsite/rst/plugins/cliconf.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+These plugins generally correspond one-to-one to network device platforms. The appropriate cliconf plugin will
+thus be automatically loaded from the network_os variable.
```
Comment (samccann) [inline docs/docsite/rst/plugins/cliconf.rst]:
```suggestion
thus be automatically loaded based on the ``network_os`` variable.
```

File: docs/docsite/rst/plugins/cliconf.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+-------------------------
+
+You can extend Ansible to support other network devices by dropping a custom plugin into the ``cliconf_plugins`` directory.
+
+.. _using_cliconf:
+
+Using cliconf plugins
+------------------------
+
+The cliconf plugin to use is determined automatically from the ansible_network_os variable. There should be no reason to override this functionality
```
Comment (samccann) [inline docs/docsite/rst/plugins/cliconf.rst]:
```suggestion
The cliconf plugin to use is determined automatically from the ``ansible_network_os`` variable. There should be no reason to override this functionality.
```

File: docs/docsite/rst/plugins/cliconf.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+Most cliconf plugins can operate without configuration. A few have additional options that can be set to imact how
```
Comment (samccann) [inline docs/docsite/rst/plugins/cliconf.rst]:
```suggestion
Most cliconf plugins can operate without configuration. A few have additional options that can be set to impact how
```

File: docs/docsite/rst/plugins/cliconf.rst [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+tasks are translated into cli commands.
```
Comment (samccann) [inline docs/docsite/rst/plugins/cliconf.rst]:
```suggestion
tasks are translated into CLI commands.
```

File: docs/docsite/rst/plugins/cliconf.rst [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+Plugins are self-documenting. Each plugin should document its configuration options.
+
+.. _cliconf_plugin_list:
+
+Plugin List
```
Comment (samccann) [inline docs/docsite/rst/plugins/cliconf.rst]:
```suggestion
Plugin list
```

File: docs/docsite/rst/plugins/httpapi.rst [hunk 8]
```diff
@@ -0,0 +1,57 @@
+.. _httpapi_plugins:
+
+httpapi Plugins
+===============
+
+.. contents::
+   :local:
+   :depth: 2
+
+Httpapi plugins inform Ansible how to interact with a remote device's HTTP-based API and execute tasks on the
```
Comment (samccann) [inline docs/docsite/rst/plugins/httpapi.rst]:
```suggestion
Httpapi plugins tell Ansible how to interact with a remote device's HTTP-based API and execute tasks on the
```

File: docs/docsite/rst/plugins/httpapi.rst [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+device.
+
+Each plugin represents a particular dialect of API. Some are platform-specific (eos, nxos), while others
+might be usable on a variety of platforms (restconf).
```
Comment (samccann) [inline docs/docsite/rst/plugins/httpapi.rst]: Not sure which way you want to show this.  Either it should be Arista EOS, Cisco NX-OS..and RESTCONF... or put them as ``eos`` etc so it's obvious they are settings.

File: docs/docsite/rst/plugins/httpapi.rst [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+-------------------------
+
+You can extend Ansible to support other APIs by dropping a custom plugin into the ``httpapi_plugins`` directory.
+
+.. _using_httpapi:
+
+Using httpapi plugins
+------------------------
+
+The httpapi plugin to use is determined automatically from the ansible_network_os variable.
```
Comment (samccann) [inline docs/docsite/rst/plugins/httpapi.rst]:
```suggestion
The httpapi plugin to use is determined automatically from the ``ansible_network_os`` variable.
```

File: docs/docsite/rst/plugins/cliconf.rst (hunk 7 above)
Comment (Qalthos) [reply docs/docsite/rst/plugins/cliconf.rst]: Pretty sure this should still be title case. In any case, this is how it is in all other pages

Comment (samccann) [reply docs/docsite/rst/plugins/cliconf.rst]: not a biggie. We're slowly moving to sentence case so I comment as I notice it in reviews...

File: lib/ansible/plugins/cliconf/aireos.py [hunk 11]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: aireos
+short_description: Use aireos cliconf to run command on aireos platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/aireos.py]:
```suggestion
short_description: Use aireos cliconf to run command on Cisco WLC platform
```

File: lib/ansible/plugins/cliconf/aireos.py [hunk 12]
```diff
@@ -19,6 +19,16 @@
...
+description:
+  - This aireos plugin provides low level abstraction apis for
+    sending and receiving CLI commands from aireos network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/aireos.py]:
```suggestion
    sending and receiving CLI commands from AireOS network devices.
```

File: lib/ansible/plugins/cliconf/aruba.py [hunk 13]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: aruba
+short_description: Use aruba cliconf to run command on aruba platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/aruba.py]:
```suggestion
short_description: Use aruba cliconf to run command on Aruba platform
```

File: lib/ansible/plugins/cliconf/aruba.py [hunk 14]
```diff
@@ -19,6 +19,16 @@
...
+description:
+  - This aruba plugin provides low level abstraction apis for
+    sending and receiving CLI commands from aruba network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/aruba.py]:
```suggestion
    sending and receiving CLI commands from Aruba network devices.
```

File: lib/ansible/plugins/cliconf/asa.py [hunk 15]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: asa
+short_description: Use asa cliconf to run command on asa platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/asa.py]:
```suggestion
short_description: Use asa cliconf to run command on Cisco ASA platform
```

File: lib/ansible/plugins/cliconf/asa.py [hunk 16]
```diff
@@ -19,6 +19,16 @@
...
+description:
+  - This asa plugin provides low level abstraction apis for
+    sending and receiving CLI commands from asa network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/asa.py]:
```suggestion
    sending and receiving CLI commands from Cisco ASA network devices.
```

File: lib/ansible/plugins/cliconf/ce.py [hunk 17]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: ce
+short_description: Use ce cliconf to run command on ce platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/ce.py]:
```suggestion
short_description: Use ce cliconf to run command on HUAWEI CloudEngine platform
```

File: lib/ansible/plugins/cliconf/ce.py [hunk 18]
```diff
@@ -19,6 +19,16 @@
...
+description:
+  - This ce plugin provides low level abstraction apis for
+    sending and receiving CLI commands from ce network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/ce.py]:
```suggestion
    sending and receiving CLI commands from HUAWEI CloudEngine network devices.
```

File: lib/ansible/plugins/cliconf/cnos.py [hunk 19]
```diff
@@ -16,6 +16,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: cnos
+short_description: Use cnos cliconf to run command on cnos platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/cnos.py]:
```suggestion
short_description: Use cnos cliconf to run command on Lenovo CNOS platform
```

File: lib/ansible/plugins/cliconf/cnos.py [hunk 20]
```diff
@@ -16,6 +16,16 @@
...
+description:
+  - This cnos plugin provides low level abstraction apis for
+    sending and receiving CLI commands from cnos network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/cnos.py]:
```suggestion
    sending and receiving CLI commands from Lenovo CNOS network devices.
```

File: lib/ansible/plugins/cliconf/dellos10.py [hunk 21]
```diff
@@ -21,6 +21,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: dellos10
+short_description: Use dellos10 cliconf to run command on dellos10 platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/dellos10.py]:
```suggestion
short_description: Use dellos10 cliconf to run command on Dell OS10 platform
```

File: lib/ansible/plugins/cliconf/dellos10.py [hunk 22]
```diff
@@ -21,6 +21,16 @@
...
+description:
+  - This dellos10 plugin provides low level abstraction apis for
+    sending and receiving CLI commands from dellos10 network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/dellos10.py]:
```suggestion
    sending and receiving CLI commands from Dell OS10 network devices.
```

File: lib/ansible/plugins/cliconf/dellos6.py [hunk 23]
```diff
@@ -21,6 +21,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: dellos6
+short_description: Use dellos6 cliconf to run command on dellos6 platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/dellos6.py]:
```suggestion
short_description: Use dellos6 cliconf to run command on Dell OS6 platform
```

File: lib/ansible/plugins/cliconf/dellos6.py [hunk 24]
```diff
@@ -21,6 +21,16 @@
...
+description:
+  - This dellos6 plugin provides low level abstraction apis for
+    sending and receiving CLI commands from dellos6 network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/dellos6.py]:
```suggestion
    sending and receiving CLI commands from Dell OS6 network devices.
```

File: lib/ansible/plugins/cliconf/dellos9.py [hunk 25]
```diff
@@ -21,6 +21,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: dellos9
+short_description: Use dellos9 cliconf to run command on dellos9 platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/dellos9.py]:
```suggestion
short_description: Use dellos9 cliconf to run command on Dell OS9 platform
```

File: lib/ansible/plugins/cliconf/dellos9.py [hunk 26]
```diff
@@ -21,6 +21,16 @@
...
+description:
+  - This dellos9 plugin provides low level abstraction apis for
+    sending and receiving CLI commands from dellos9 network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/dellos9.py]:
```suggestion
    sending and receiving CLI commands from Dell OS9 network devices.
```

File: lib/ansible/plugins/cliconf/edgeos.py [hunk 27]
```diff
@@ -5,6 +5,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: edgeos
+short_description: Use edgeos cliconf to run command on edgeos platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/edgeos.py]:
```suggestion
short_description: Use edgeos cliconf to run command on EdgeOS platform
```

File: lib/ansible/plugins/cliconf/edgeswitch.py [hunk 28]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: edgeswitch
+short_description: Use edgeswitch cliconf to run command on edgeswitch platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/edgeswitch.py]:
```suggestion
short_description: Use edgeswitch cliconf to run command on EdgeSwitch platform
```

File: lib/ansible/plugins/cliconf/enos.py [hunk 29]
```diff
@@ -16,6 +16,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: enos
+short_description: Use enos cliconf to run command on enos platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/enos.py]:
```suggestion
short_description: Use enos cliconf to run command on Lenovo ENOS platform
```

File: lib/ansible/plugins/cliconf/enos.py [hunk 30]
```diff
@@ -16,6 +16,16 @@
...
+description:
+  - This enos plugin provides low level abstraction apis for
+    sending and receiving CLI commands from enos network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/enos.py]:
```suggestion
    sending and receiving CLI commands from Lenovo ENOS network devices.
```

File: lib/ansible/plugins/cliconf/eos.py [hunk 31]
```diff
@@ -25,9 +25,9 @@
 cliconf: eos
 short_description: Use eos cliconf to run command on eos platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/eos.py]:
```suggestion
short_description: Use eos cliconf to run command on Arista EOS platform
```

File: lib/ansible/plugins/cliconf/exos.py [hunk 32]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: exos
+short_description: Use exos cliconf to run command on exos platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/exos.py]:
```suggestion
short_description: Use exos cliconf to run command on Extreme EXOS platform
```

File: lib/ansible/plugins/cliconf/exos.py [hunk 33]
```diff
@@ -19,6 +19,16 @@
...
+description:
+  - This exos plugin provides low level abstraction apis for
+    sending and receiving CLI commands from exos network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/exos.py]:
```suggestion
    sending and receiving CLI commands from Extreme EXOS network devices.
```

File: lib/ansible/plugins/cliconf/frr.py [hunk 34]
```diff
@@ -25,8 +25,8 @@
 cliconf: frr
 short_description: Use frr cliconf to run command on Free Range Routing platform
 description:
-  - This frr plugin provides low level abstraction api's for
-    sending and receiving CLI commands from devices running frr.
+  - This frr plugin provides low level abstraction apis for
+    sending and receiving CLI commands from frr network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/frr.py]:
```suggestion
    sending and receiving CLI commands from FRR network devices.
```

File: lib/ansible/plugins/cliconf/ios.py [hunk 35]
```diff
@@ -19,6 +19,17 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+author: Ansible Networking Team
+cliconf: ios
+short_description: Use ios cliconf to run command on ios platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/ios.py]:
```suggestion
short_description: Use ios cliconf to run command on Cisco IOS platform
```

File: lib/ansible/plugins/cliconf/iosxr.py [hunk 36]
```diff
@@ -19,6 +19,17 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+author: Ansible Networking Team
+cliconf: iosxr
+short_description: Use iosxr cliconf to run command on iosxr platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/iosxr.py]:
```suggestion
short_description: Use iosxr cliconf to run command on Cisco IOS XR platform
```

File: lib/ansible/plugins/cliconf/iosxr.py [hunk 37]
```diff
@@ -19,6 +19,17 @@
...
+description:
+  - This iosxr plugin provides low level abstraction apis for
+    sending and receiving CLI commands from Cisco IOS-XR network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/iosxr.py]:
```suggestion
    sending and receiving CLI commands from Cisco IOS XR network devices.
```

File: lib/ansible/plugins/cliconf/ironware.py [hunk 38]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: ironware
+short_description: Use ironware cliconf to run command on ironware platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/ironware.py]:
```suggestion
short_description: Use ironware cliconf to run command on Extreme Ironware platform
```

File: lib/ansible/plugins/cliconf/ironware.py [hunk 39]
```diff
@@ -19,6 +19,16 @@
...
+description:
+  - This ironware plugin provides low level abstraction apis for
+    sending and receiving CLI commands from ironware network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/ironware.py]:
```suggestion
    sending and receiving CLI commands from Extreme Ironware network devices.
```

File: lib/ansible/plugins/cliconf/junos.py [hunk 40]
```diff
@@ -19,6 +19,17 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+author: Ansible Networking Team
+cliconf: junos
+short_description: Use junos cliconf to run command on Juniper platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/junos.py]:
```suggestion
short_description: Use junos cliconf to run command on Juniper Junos OS platform
```

File: lib/ansible/plugins/cliconf/nos.py [hunk 41]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: nos
+short_description: Use nos cliconf to run command on nos platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/nos.py]:
```suggestion
short_description: Use nos cliconf to run command on Extreme NOS platform
```

File: lib/ansible/plugins/cliconf/nos.py [hunk 42]
```diff
@@ -19,6 +19,16 @@
...
+description:
+  - This nos plugin provides low level abstraction apis for
+    sending and receiving CLI commands from nos network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/nos.py]:
```suggestion
    sending and receiving CLI commands from Extreme NOS network devices.
```

File: lib/ansible/plugins/cliconf/nxos.py [hunk 43]
```diff
@@ -19,6 +19,17 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+author: Ansible Networking Team
+cliconf: nxos
+short_description: Use nxos cliconf to run command on nxos platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/nxos.py]:
```suggestion
short_description: Use nxos cliconf to run command on Cisco NX-OS platform
```

File: lib/ansible/plugins/cliconf/onyx.py [hunk 44]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: onyx
+short_description: Use onyx cliconf to run command on onyx platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/onyx.py]:
```suggestion
short_description: Use onyx cliconf to run command on Mellanox ONYX platform
```

File: lib/ansible/plugins/cliconf/onyx.py [hunk 45]
```diff
@@ -19,6 +19,16 @@
...
+description:
+  - This onyx plugin provides low level abstraction apis for
+    sending and receiving CLI commands from onyx network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/onyx.py]:
```suggestion
    sending and receiving CLI commands from Mellanox ONYX network devices.
```

File: lib/ansible/plugins/cliconf/routeros.py [hunk 46]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: routeros
+short_description: Use roteros cliconf to run command on roteros platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/routeros.py]:
```suggestion
short_description: Use roteros cliconf to run command on MikroTik RouterOS platform
```

File: lib/ansible/plugins/cliconf/routeros.py [hunk 47]
```diff
@@ -19,6 +19,16 @@
...
+description:
+  - This routeros plugin provides low level abstraction apis for
+    sending and receiving CLI commands from routeros network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/routeros.py]:
```suggestion
    sending and receiving CLI commands from MikroTik RouterOS network devices.
```

File: lib/ansible/plugins/cliconf/slxos.py [hunk 48]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: slxos
+short_description: Use slxos cliconf to run command on slxos platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/slxos.py]:
```suggestion
short_description: Use slxos cliconf to run command on Extreme SLX-OS platform
```

File: lib/ansible/plugins/cliconf/slxos.py [hunk 49]
```diff
@@ -19,6 +19,16 @@
...
+description:
+  - This slxos plugin provides low level abstraction apis for
+    sending and receiving CLI commands from slxos network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/slxos.py]:
```suggestion
    sending and receiving CLI commands from Extreme SLX-OS network devices.
```

File: lib/ansible/plugins/cliconf/voss.py [hunk 50]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: voss
+short_description: Use voss cliconf to run command on voss platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/voss.py]:
```suggestion
short_description: Use voss cliconf to run command on Extreme VOSS platform
```

File: lib/ansible/plugins/cliconf/voss.py [hunk 51]
```diff
@@ -19,6 +19,16 @@
...
+description:
+  - This voss plugin provides low level abstraction apis for
+    sending and receiving CLI commands from voss network devices.
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/voss.py]:
```suggestion
    sending and receiving CLI commands from Extreme VOSS network devices.
```

File: lib/ansible/plugins/cliconf/vyos.py [hunk 52]
```diff
@@ -19,6 +19,17 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+author: Ansible Networking Team
+cliconf: vyos
+short_description: Use vyos cliconf to run command on vyos platform
```
Comment (samccann) [inline lib/ansible/plugins/cliconf/vyos.py]:
```suggestion
short_description: Use vyos cliconf to run command on VyOS platform
```

**Review by samccann — COMMENTED** [review]
Seems like a lot, but it's just putting the full network (os) name and capitalization in each of the plugin descriptions (so they match the modules)

**Comment by Qalthos** [comment]
> Seems like a lot, but it's just putting the full network (os) name and capitalization in each of the plugin descriptions (so they match the modules)

Yeah, I wasn't entirely sure what most of them were, so I just went for platform name. Thanks!

File: lib/ansible/plugins/cliconf/eos.py (hunk 31 above)
Comment (Qalthos) [reply lib/ansible/plugins/cliconf/eos.py]: Looks like there's an extra space inf front here?

File: lib/ansible/plugins/cliconf/routeros.py [hunk 53]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: routeros
+short_description: Use roteros cliconf to run command on MikroTik RouterOS platform
```
Comment (Qalthos) [inline lib/ansible/plugins/cliconf/routeros.py]:
```suggestion
short_description: Use routeros cliconf to run command on MikroTik RouterOS platform
```

File: lib/ansible/plugins/cliconf/aireos.py [hunk 54]
```diff
@@ -19,6 +19,16 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+DOCUMENTATION = """
+---
+cliconf: aireos
+short_description: Use aireos cliconf to run command on Cisco WLC platform
```
Comment (Qalthos) [inline lib/ansible/plugins/cliconf/aireos.py:25]: These don't match? Should we pick one?

Edit: AireOS is used below

Comment (samccann) [reply lib/ansible/plugins/cliconf/aireos.py:25]: looks like the modules both use Cisco WLC so lets go with that unless you thing AireOS is more well-known.

Comment (Qalthos) [reply lib/ansible/plugins/cliconf/aireos.py:25]: :+1:

File: lib/ansible/plugins/cliconf/aireos.py [hunk 55]
```diff
@@ -19,6 +19,16 @@
...
+description:
+  - This aireos plugin provides low level abstraction apis for
+    sending and receiving CLI commands from AireOS network devices.
```
Comment (Qalthos) [inline lib/ansible/plugins/cliconf/aireos.py]:
```suggestion
    sending and receiving CLI commands from Cisco WLC network devices.
```

**Pull request merged**